firebase-database/src/androidTest/java/com/google/firebase/database/QueryTest.java

```diff
@@ -504,6 +585,37 @@ public void setVariousLimitsWithStartAtNameWithServerData()
     expectations.waitForEvents();
   }
 
+  @Test
+  public void setVariousLimitsWithStartAfterNameWithServerData()
+      throws DatabaseException, InterruptedException, TestFailure, ExecutionException,
+          TimeoutException {
+    DatabaseReference ref = IntegrationTestHelpers.getRandomNode();
+
+    // TODO: this test kinda has race conditions. The listens are added sequentially, so we get a
+    // lot of partial data back from the server. This all correct, and we end up in the correct
+    // state, but it's still kinda weird. Consider having ValueExpectationHelper deal with initial
+    // state.
+
+    new WriteFuture(ref, new MapBuilder().put("a", 1).put("b", 2).put("c", 3).put("d", 4).build())
+        .timedGet();
+
+    ValueExpectationHelper expectations = new ValueExpectationHelper();
+
+    expectations.add(ref.startAfter(null).limitToFirst(1), null);
```

Comment:
Let's talk about this offline. My gut feeling is that this should match all results, since they order past `null`.

Reply:
I added an explanation here: https://github.com/firebase/firebase-ios-sdk/pull/7209/files#r556060181. 

I know this reads funny, but I think it is the intended behavior because we are indexing on priority.